src/test/java/kitchenpos/application/MenuGroupServiceTest.java

```diff
@@ -30,11 +31,13 @@ class MenuGroupServiceTest {
     void createTest() {
         // given
         MenuGroup request = TEST_MENU_GROUP();
+        given(menuGroupRepository.save(any(MenuGroup.class))).willReturn(request);
         // when
-        menuGroupService.create(request);
+        MenuGroup actual = menuGroupService.create(request);
 
         // then
         then(menuGroupRepository).should(times(1)).save(any(MenuGroup.class));
+        assertThat(actual).isEqualTo(request);
```

Comment:
👍 등록 성공에 대한 검증은 실제 요청한 데이터로 등록이 잘 되었나로 생각되어어죠.
그래서 어떤 메서드를 호출했는지 보다는 해당 기능에 대한 검증이 더 명확하고 좋네요.
추가로, 현재는 동일한 객체를 비교하는 것이기 때문에 실제 잘 적용된 것인지 위의 코드보다는 검증쪽은 실제 값을 넣어주는 것이 좋아보여요.

```suggestion
assertThat(actual.getId()l).isNotNull();
assertThat(actual.getName()).isEqualTo("메뉴그룹이름");
```

Reply:
객체간 비교하는 것보다는 비교해야하는 값을 비교하는 것이 더 명시적으로 보이네요!

Reply:
말씀해주신 경우는 동일성 비교일 것 같아요.

객체에 Equals 코드를 오버라이드해서 값비교로 하지 않았다면, 객체의 주소를 확인할 것 같아서요./

실제로 실무에서 객체간 동일성 비교를 하기 보다는 ID값으로 비교를 많이 하는 듯 합니다.

Reply:
네 맞아요. 💯  equals 를 재정의해주지 않았기 때문에 최상위 객체인 Object 의 equals 를 사용하게 되는데 이때 동일성 비교를해요.
즉, 주소값을 비교하기 때문에 원하는 검증이 안될 수 있어요.
적절하게 equals 메서드를 재정의해주어서 객체 간의 동등성 비교도 많이 사용해보세요 👍